fastplotlib/widgets/image.py

```diff
@@ -615,61 +483,42 @@ def window_funcs(self) -> Dict[str, _WindowFunctions]:
         return self._window_funcs
 
     @window_funcs.setter
-    def window_funcs(self, sa: Union[int, Dict[str, int]]):
+    def window_funcs(self, sa: Dict[str, int]):
         if sa is None:
             self._window_funcs = None
             # force frame to update
             self.current_index = self.current_index
             return
 
-        # for a single dim
-        elif isinstance(sa, tuple):
-            if len(self.slider_dims) > 1:
-                raise TypeError(
-                    "Must pass dict argument to window_funcs if using multiple sliders. See the docstring."
-                )
-            if not callable(sa[0]) or not isinstance(sa[1], int):
-                raise TypeError(
-                    "Tuple argument to `window_funcs` must be in the form of (func, window_size). See the docstring."
-                )
-
-            dim_str = self.slider_dims[0]
-            self._window_funcs = dict()
-            self._window_funcs[dim_str] = _WindowFunctions(self, *sa)
-
-        # for multiple dims
         elif isinstance(sa, dict):
+            if not set(sa.keys()).issubset(ALLOWED_WINDOW_KEYS):
+                raise ValueError(f"The only allowed keys to window funcs are {list(ALLOWED_WINDOW_KEYS)} " 
+                                 f"Your window func passed in these keys: {list(sa.keys())}")
             if not all(
-                [isinstance(_sa, tuple) or (_sa is None) for _sa in sa.values()]
+                [isinstance(_sa, tuple) for _sa in sa.values()]
             ):
                 raise TypeError(
                     "dict argument to `window_funcs` must be in the form of: "
                     "`{dimension: (func, window_size)}`. "
                     "See the docstring."
                 )
             for v in sa.values():
-                if v is not None:
-                    if not callable(v[0]) or not (
-                        isinstance(v[1], int) or v[1] is None
-                    ):
-                        raise TypeError(
-                            "dict argument to `window_funcs` must be in the form of: "
-                            "`{dimension: (func, window_size)}`. "
-                            "See the docstring."
-                        )
+                if not callable(v[0]):
```

Comment:
should also check that `v[1] is int` right?

Reply:
@kushalkolar good call